sound/soc/sof/core.c

```diff
@@ -461,17 +461,13 @@ int snd_sof_device_remove(struct device *dev)
 	if (IS_ENABLED(CONFIG_SND_SOC_SOF_PROBE_WORK_QUEUE))
 		cancel_work_sync(&sdev->probe_work);
 
-	snd_soc_unregister_component(dev);
 	snd_sof_fw_unload(sdev);
 	snd_sof_ipc_free(sdev);
 	snd_sof_free_debug(sdev);
 	snd_sof_free_trace(sdev);
 	snd_sof_remove(sdev);
-	/*
-	 * platform_device_unregister() frees the card and its resources.
-	 * So it should be called after unregistering the comp driver
-	 * so that the card is valid while unregistering comp driver.
-	 */
```

Comment:
isn't this comment still relevant, even with the use of devm_ ? does it need to be removed completely or reworded?